src/blog/0.2/version-0.2-released.vmd

```diff
@@ -0,0 +1,443 @@
+
+
+For a real-world use case, check out the recent article [Higher RAII, the pattern that saved me a vital 5 hours in the 7DRL Challenge](https://verdagon.dev/blog/higher-raii-7drl).
+
+
+<<<<
+To use Higher RAII, just add a `#!DeriveStructDrop` above your struct or `#!DeriveInterfaceDrop` above your interface, to prevent it from automatically defining a `drop` function.
+
+
+Instead of having an automatically-called `drop` function, this example has a `destroyShip` function that takes a boolean parameter and returns an integer.
+
+Since there is no `drop` function for the struct, Vale will never automatically destroy the object. The user _must_ call this `destroyShip` function.
+
+
+This pattern can be used to enforce that we don't forget to call a certain function (like `destroyShip`) at some point in the future.
+////
+```vale
+#!DeriveStructDrop
+struct Spaceship {
+  name str;
+  fuel int;
+}
+
+exported func main() {
+  ship = Spaceship("Serenity", 2);
+  // ship is an owning reference
+
+  println(ship.name);
+
+  fuel = (ship).destroyShip(true);
+  println("Fuel was {fuel}.")
+}
+
+func destroyShip(
+  s Spaceship,
+  print bool)
+int {
+  [name, fuel] = s; // Deallocates ship
+  if print {
+    println("Destroyed {name}!");
+  }
+  return fuel;
+}
+```stdout
+Destroyed Serenity!
+Fuel was 2.
+```
+>>>>
+
+
+
+Higher RAII can be also be used for more powerful [type-state programming](http://cliffle.com/blog/rust-typestate/) [#tsp] than any other imperative language. Try it out!
+
+
+
+<slice>
+#longer: We're particularly excited to prototype the [region borrow checker](/blog/zero-cost-refs-regions), which could help Vale approach C++'s speed by getting rid of the vast majority of our memory safety overhead, and enable concurrency features like described in [Seamless, Fearless, Structured Concurrency](/blog/seamless-fearless-structured-concurrency).
+
+We've been looking forward to this version for a long time!
+
+
+#tsp: Type-state programming is a way to use the compiler to ensure that specific functions are only callable when the object is in a certain state. This is all tracked via the type system, with zero run-time overhead.
+</slice>
+
+
+# Concept Functions
+
+Vale now supports *concept functions*, a way to specify that certain functions must exist for given generic parameters, without making them implement any traits.
+
+
+For example, List's `clone` function can require that there exists a `clone` function for its elements:
+
+```vale
+func clone<T>(self &List<T>) List<T>
+    where func clone(&T)T
+{
+  result = List<T>();
+  foreach x in self {
+    result.add(clone(x));
+  }
+  return result;
+}
+```
+
+
+
+This is often much easier than what we normally see in mainstream languages, which is to require the element to implement a certain trait.
```

Comment:
Its weird to mention traits here cause it assumes they are the status quo.
Instead just say "which is to require an explicit implementation for the type"